test/unit/org/apache/cassandra/service/accord/AccordCommandStoreFuzzTest.java

```diff
@@ -0,0 +1,457 @@
+    {
+        return assertPreAccept(instance, txn, route, keyConflicts, keys, Collections.emptyList(), Ranges.EMPTY);
+    }
+
+    private static TxnId assertPreAccept(Instance instance,
+                                         Txn txn, FullRoute<?> route,
+                                         List<TxnId> keyConflicts, Keys keys,
+                                         List<TxnId> rangeConflicts, Ranges ranges) throws ExecutionException, InterruptedException
+    {
+        var pair = assertPreAcceptAsync(instance, txn, route, keyConflicts, keys, rangeConflicts, ranges);
+        instance.processAll();
+        AsyncChains.getBlocking(pair.right);
+
+        return pair.left;
+    }
+
+    private static Pair<TxnId, AsyncResult<?>> assertPreAcceptAsync(Instance instance,
+                                                                      Txn txn, FullRoute<?> route,
+                                                                      List<TxnId> keyConflicts, Keys keys)
+    {
+        return assertPreAcceptAsync(instance, txn, route, keyConflicts, keys, Collections.emptyList(), Ranges.EMPTY);
+    }
+
+    private static Pair<TxnId, AsyncResult<?>> assertPreAcceptAsync(Instance instance,
+                                                                      Txn txn, FullRoute<?> route,
+                                                                      List<TxnId> keyConflicts, Keys keys,
+                                                                      List<TxnId> rangeConflicts, Ranges ranges)
+    {
+        int kConflictSize = keyConflicts.size();
+        int rConflictSize = rangeConflicts.size();
+        TxnId txnId = new TxnId(instance.timeService.epoch(), instance.timeService.now(), txn.kind(), txn.keys().domain(), nodeId);
+        var preAccept = new PreAccept(instance.nodeId, new Topologies.Single(SizeOfIntersectionSorter.SUPPLIER, instance.topology), txnId, txn, route);
+        return Pair.create(txnId, instance.processAsync(preAccept, safe -> {
+            var reply = preAccept.apply(safe);
+            Assertions.assertThat(reply.isOk()).isTrue();
+            return (PreAccept.PreAcceptOk) reply;
+        }).map(success -> {
+            assertDeps(keyConflicts.subList(0, kConflictSize), keys, rangeConflicts.subList(0, rConflictSize), ranges, success);
+            return null;
+        }).beginAsResult());
+    }
+
+    private static void assertDeps(List<TxnId> keyConflicts, Keys keys,
+                                   List<TxnId> rangeConflicts, Ranges ranges,
+                                   PreAccept.PreAcceptOk success)
+    {
+        if (rangeConflicts.isEmpty())
+        {
+            Assertions.assertThat(success.deps.rangeDeps.isEmpty()).describedAs("rangeDeps was not empty").isTrue();
+        }
+        else
+        {
+            Assertions.assertThat(success.deps.rangeDeps.rangeCount()).describedAs("Expected ranges size").isEqualTo(ranges.size());
```

Comment:
these asserts are not touched with these tests, they are used in the range tests which are left out of this patch due to range not actually working right now... my range patch gets this all passing.